codegen/codegen_test.go

```diff
@@ -0,0 +1,159 @@
+package codegen
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/moby/buildkit/client/llb"
+	"github.com/openllb/hlb/checker"
+	"github.com/openllb/hlb/parser"
+	"github.com/openllb/hlb/solver"
+	"github.com/stretchr/testify/require"
+	"github.com/xlab/treeprint"
+)
+
+func Expect(st llb.State) solver.Request {
```

Comment:
I think we might want to make this `func Expect(t *testing.T, st llb.State)` so that we can `require.NoError(t, err)` in the code. The panics will be hard to debug which test failed otherwise.